File: JavaScript/8js 3/weatherInfo.js
/*
|--------------------------------------------------------------------------
| Grasshopper - Debug
|--------------------------------------------------------------------------
|
*/

const convertToCelsius = f => (f - 32) * (5 / 9)
function weatherInfo(temp) {
   const c = convertToCelsius(temp)
   return c > 0 ? `${c} is above freezing temperature` : `${c} is freezing temperature`
}


function weatherInfo(temp) {
   return `${temp = (temp - 32) * (5 / 9)} is ${temp > 0 ? 'above ': ''}freezing temperature`;
}


function weatherInfo(temp) {
   const c = (temp - 32) * (5 / 9)
   return (c < 0) ? (c + " is freezing temperature") : (c + " is above freezing temperature")
}


console.log(weatherInfo(50), '10 is above freezing temperature')
console.log(weatherInfo(23), '-5 is freezing temperature')

File: Python/8py 3/weather_info.py
"""
|--------------------------------------------------------------------------
| Grasshopper - Debug
|--------------------------------------------------------------------------
|
"""

def weather_info (temp):
   temp = (temp - 32) * (5 / 9)
   return f'{temp} is above freezing temperature' if temp > 0 else f'{temp} is freezing temperature'


def weather_info(fahrenheit):
   celsius = 5.0 * (fahrenheit - 32) / 9
   return "{} is {}freezing temperature".format(celsius, "above " if celsius >= 0 else "")


def weather_info(f_temp):
   c_temp = (f_temp - 32) * (5/9)
   return '{} is {}freezing temperature'.format(c_temp, 'above ' if c_temp > 0 else '')
        

print(weather_info(50), '10.0 is above freezing temperature')
print(weather_info(23),  '-5.0 is freezing temperature')